### PR TITLE
Fix javascript-apollo generator/template

### LINKED ISSUE #13190: [BUG] Generator javascript-apollo is outdated - generated code does not compile
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

The template uses the version 0.7.0 of the [`apollo-data-source` package](https://www.npmjs.com/package/apollo-datasource-rest), which is quite old (3 years old).
The library has since changed considerably, and I believe users of this generator should enjoy the latest features.

The generated code is incompatible with the latest version (currently 3.6.1), hence this bug report.
Additionally, I was not able to compile the code using the old version of `apollo-data-source`.

The generator as-is, seems to be badly broken and unusable (but do feel free to correct me!)

##### openapi-generator version

Version 6.0.1; not a regression

##### OpenAPI declaration file content or url

I'm using [EVE Online's ESI](https://esi.evetech.net/latest/swagger.json) for my purposes.
Can try to come up with a minimal complete example if required/asked.

##### Generation Details

```
openapi-generator generate -g javascript-apollo -i https://esi.evetech.net/latest/swagger.json -o src/generated/esi-latest-custom-template
```

##### Steps to reproduce

The generated code does not compile.

##### Related issues/PRs

Didn't find any

##### Suggest a fix

Will open PR

### PR DESCRIPTION
**Pull request by joaomlneto**
Generated code does not compile with recent `apollo-server` versions. This PR attempts to make it generate compatible code with latest versions of the Apollo libraries.

Comments/reviews are highly desired! I do not fully understand the code, but my changes appear to fix compilation issues.

I'm submitting it to the `master` branch, but may be more appropriate to submit it to another branch?

Additionally, this may break compatibility with potential users using very old versions. What would be the best course of action to tackle this?

NodeJS/Javascript technical committee: @CodeNinjai @frol @cliffano 

Closes #13190 

### Todo list:
- [x] Bump Apollo Server compatibility to version 3
- [x] Fix tests
- [x] Fix parameters pased onto callApi — parameters `contentTypes` and `accepts` are missing.
- [x]  https://github.com/apollographql/apollo-server/pull/1277 — make `requestInit` parameter available in every endpoint, passed onto `callApi`, subsequently passed onto the `get`/`post`/`delete`/`patch`/`put` methods on `RESTDataSource`.

### Resources
- Previous code included `apollo-datasource-rest` library version 0.7.0. [Browse `apollo-datasource-rest@v0.7.0`](https://github.com/apollographql/apollo-server/tree/0743d6b2f1737758cf09e80d2086917772bc00c9/packages/apollo-datasource-rest)

---

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
@joaomlneto thanks for the PR. When you've time, can you please PM me via Slack to discuss this further?

https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

**Comment by joaomlneto** [comment]
@wing328 — got to it earlier than expected!

There's still a few things that can definitely be improved (not passing `contentType` and `accepts` request headers yet — will work on it!), but I think this is already plenty for a first PR.

The code generated compiles successfully 🎉 and can be tested (although each test method is trivially empty for now — haven't looked at the other generators to see what's your standard here):
```bash
npm install && npm test
```

I'll keep using the template and will contribute to it in the coming weeks. 😄 

Thanks for OpenAPI Generator!

File: modules/openapi-generator/src/main/resources/Javascript-Apollo/ApiClient.mustache [hunk 1]
```diff
@@ -53,7 +53,7 @@ export default class ApiClient extends RESTDataSource {
     }
 
     parametrizePath(path, pathParams) {
-        return url.replace(/\{([\w-]+)\}/g, (fullMatch, key) => {
```
Comment (joaomlneto) [inline modules/openapi-generator/src/main/resources/Javascript-Apollo/ApiClient.mustache:56]: I have no idea what `url` referred to — there's no such variable! Switching to `path` appears to work (though it's a bit of a blind solve).

File: modules/openapi-generator/src/main/resources/Javascript-Apollo/api_test.mustache [hunk 2]
```diff
@@ -1,13 +1,15 @@
 {{>licenseInfo}}
 // CommonJS-like environments that support module.exports, like Node.
-factory(require('expect.js'), require(process.cwd()+'/src/{{#invokerPackage}}{{.}}/{{/invokerPackage}}index'));
+//factory(require('expect.js'), require(process.cwd()+'/src/{{#invokerPackage}}{{.}}/{{/invokerPackage}}index'));
```
Comment (joaomlneto) [inline modules/openapi-generator/src/main/resources/Javascript-Apollo/api_test.mustache:3]: I have no idea what this does. `factory` does not exist anywhere.

The second `require` is the inclusion of the Api being tested in this file, which is included below in a way that works (and is more human-readable/writeable)

Comment (wing328) [reply modules/openapi-generator/src/main/resources/Javascript-Apollo/api_test.mustache:3]: I think these are carried over or copied from https://github.com/OpenAPITools/openapi-generator/blob/5d68bd6a03f0c48e838b4fe3b98b7e30858c0373/modules/openapi-generator/src/main/resources/Javascript/es6/model_test.mustache#L8, in which this generator referenced.

**Pull request merged**